### ISSUE TITLE
Can this package get converted to native dart 

### ISSUE THREAD

**Issue by guyluz11** [jonathantribouharet/flutter_cast#27]
I was wondering if this package or part of it can get converted to native dart and thus work on all platforms, and can get executed from server side as well?.

What part of this library require the flutter framework and what part of it only required native dart code.

**Comment by jonathantribouharet** [comment]
The code doesn't use flutter in anyway, but it has one dependency to `bonsoir`, which is used to detect Chromecast with mDNS protocol, which has a dependency to flutter.
There are not many mDNS lib working unfortunately (I've use `mdns_plugin` then `multicast_dns` then `bonsoir`).

**Comment by guyluz11** [comment]
I see, so if Bonsoir will add dart native support it will help us.
Opened an issue for that in the Bonsoir repo
https://github.com/Skyost/Bonsoir/issues/36

I have also found [dart_chromecast](https://pub.dev/packages/dart_chromecast) that support dart native, it is interesting how did they implement detection of Chromecast (mDNS), it looks like they are using [multicast_dns](https://pub.dev/packages/multicast_dns) for that.

**Comment by jonathantribouharet** [comment]
Yes if bonsoir would support dart native, I will be able to do the same.
I used multicast_dns before but it doesn't work anymore on iOS (last check 8 Jun 2021), and from what I see the problem is still there https://github.com/flutter/flutter/issues/52733

**Comment by guyluz11** [comment]
After the issue is resolved this package will need to change name lol.

If the ios is the only problem maybe I create a fork for dart native for now :)

**Comment by guyluz11** [comment]
btw have you tested [nsd](https://pub.dev/packages/nsd)?.
I am thinking of using it for something else because it supports receiving a list of all service types in the network using _services._dns-sd._udp request.
Was wandering if it works before starting to put effort into using it.
